### ISSUE TITLE
Question on when startup gets copied to running

### ISSUE THREAD

**Issue by bbergquist0930** [sysrepo/sysrepo#1811]
Latest code build of the "devel" branchs of sysrepo, netopeer2, libyang, libssh.

When does the startup configuration get copied to the running configuration? For example, I have the "examples.yang" module loaded. The startup configuration has:

```
root@brett-VirtualBox:~# sysrepocfg -X -d startup -m examples
<cont xmlns="urn:examples">
  <l>1234</l>
</cont>
root@brett-VirtualBox:~# 
```
and the running configuration has:
```
root@brett-VirtualBox:~# sysrepocfg -X -d running -m examples
<cont xmlns="urn:examples">
  <l>4321</l>
</cont>
root@brett-VirtualBox:~# 
```
and the connection count is
```
root@brett-VirtualBox:~# sysrepoctl -C
0
root@brett-VirtualBox:~# 
```
If I start netopeer-server
```
root@brett-VirtualBox:~# netopeer2-server -d
```
And then look at the running configuration I would expect the running configuration to be the startup configuration.
```
root@brett-VirtualBox:~# sysrepocfg -X -d running -m examples
<cont xmlns="urn:examples">
  <l>4321</l>
</cont>
root@brett-VirtualBox:~# 
```
Am I missing something fundamental?

**Comment by bbergquist0930** [comment]
Could it that where in the documentation is states 

> Startup datastore is the only persistent datastore. It includes the configuration of devices at time of their boot. This configuration is copied into running when the first Sysrepo connection (shared memory) is created after system boot.
>
>Running datastore holds the current system configuration. This datastore is edited when a configuration change occured and a device should reconfigure itself. It does not persist across reboots. If desired, it can be copied into startup to rectify this.

So is this talking about "system boot" and "reboots" actually referring to Unix/Linux system boot/reboot where basically the shared memory segments would be removed?

**Comment by michalvasko** [comment]
Hi,
firstly, please choose one way of communication and do not create an issue and ask the same question on the mailing list. Then, you should have first read the documentation and I think it is quite clear there and you understood it correctly.

Regards,
Michal

**Comment by bbergquist0930** [comment]
Sorry for the email/issue here.   I really did not know which one was correct.  And honestly I have poured over the documentation and have actually gone through the code.   

Reboot "system" hit me as rebooting the "sysrepo" system.  I was not thinking that it was rebooting the host system.   After going over the code and trying to see when/where the startup was copied to the running, I found that only occurred when the shared memory segment was created.  I then started to look for where it was removed, like maybe in a normal shutdown and did not find anything.  That is kind of when it hit me that if the host was rebooted then the shared memory segments would be removed.

So it was not for a lack of looking at the code or the documentation, it was a lack of my understand what "system reboot" and thinking it was rebooting the "sysrepo" system not the host system.

I appreciate you getting back and confirming this.
